insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/LocalInsightPlugin.java

```diff
@@ -1202,36 +1195,63 @@ private void processHistoryEvents(String serial, List<HistoryEvent> historyEvent
             if (!processHistoryEvent(serial, temporaryBasals, pumpStartedEvents, historyEvent))
                 break;
         Collections.reverse(temporaryBasals);
+
         for (InsightPumpID pumpID : pumpStartedEvents) {
-            InsightPumpID stoppedEvent = databaseHelper.getPumpStoppedEvent(pumpID.pumpSerial, pumpID.timestamp);
-            if (stoppedEvent == null || stoppedEvent.eventType.equals("PumpPaused")) continue;
-            long tbrStart = stoppedEvent.timestamp + 10000;
-            TemporaryBasal temporaryBasal = new TemporaryBasal(getInjector());
-            temporaryBasal.durationInMinutes = (int) ((pumpID.timestamp - tbrStart) / 60000);
-            temporaryBasal.date = tbrStart;
-            temporaryBasal.source = Source.PUMP;
-            temporaryBasal.pumpId = pumpID.id;
-            temporaryBasal.percentRate = 0;
-            temporaryBasal.isAbsolute = false;
+            InsightPumpID stoppedEvent = insightDatabaseDao.getPumpStoppedEvent(pumpID.getPumpSerial(), pumpID.getTimestamp());
+            if (stoppedEvent != null && stoppedEvent.getEventType().equals("PumpStopped")) {             // Search if Stop event is after 15min of Pause
+                InsightPumpID pauseEvent = insightDatabaseDao.getPumpStoppedEvent(pumpID.getPumpSerial(), stoppedEvent.getTimestamp() - T.mins(1).msecs());
+                if (pauseEvent != null && pauseEvent.getEventType().equals("PumpPaused") && (stoppedEvent.getTimestamp() - pauseEvent.getTimestamp() < T.mins(16).msecs())) {
```

Comment:
@TebbeUbben I added this to include the "Pause" in the zero temp if the Stop event is after 15 min of pause...

Reply:
Hmm, the relevant pause entry might still not be in the database if it's processed afterwards. Remember, we are processing history entries backwards for aligning the timestamps.

Reply:
We are processing backwards, but all history entries are processed (so in database) when we start to loop on `pumpStartedEvents` to set 0 temp if necessary...
I made 2 tests (one with only Stop and Start events, and another with 15min of pause before Stop, for me it works...